lib/services/ngsi/entities-NGSI-v2.js

```diff
@@ -402,6 +416,13 @@ function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, ca
                     } else {
                         delete result.id;
                         delete result.type;
+                        // Remove id, type and any hidden attrs after processing
+                        if (typeInformation.explicitAttrs) {
+                            const hidden = _.difference(_.keys(result), explicitAttrsList);
+                            hidden.forEach((attr) => {
+                                delete result[attr];
+                            });
```

Comment:
That worked fine for `hidden`, but the logic in`explicitAttrs` is the other way up. What I need to remove is elements in set B which are not in set A, so first I have to define "set A " using `typeInformation` (in one loop) and then only run the delete on the diff. (the keys of "set B" minus the keys of "set A").  The logic is now nicely encapsulated into a function so you could amend it yourself if you feel it really necessary.

Reply:
After discussing with the rest of the team in other channels, I raised a new issue, #1168 for discussing the idea of removing underscore from codebase in the future. This will not block this PR.

NTC